### PR TITLE
Bar 1125

### PR DESCRIPTION
**Pull request by Komalpreet12**

### CODE REVIEW

File: src/app/shared/directives/number-only.ts [hunk 1]
```diff
@@ -6,36 +6,10 @@ export class NumbersOnlyDirective {
...
-            (keyCode >= 35 && keyCode <= 38)) {
-            // let it happen, don't do anything
-            return;
-        }
-        // Ensure that it is a number and stop the keypress
-        if ((e.shiftKey || (keyCode < 48 || keyCode > 57))) {
-            e.preventDefault();
-        }
+    omit_special_char(event) {
+        let k;
```
Comment (jorawarsingh) [inline src/app/shared/directives/number-only.ts]: and dups here again?

**Comment by jorawarsingh** [comment]
fix issues mentioned in comments and do new pull request.

File: src/app/guides/contingent-liability-guide/step-3/step-three-component.ts [hunk 2]
```diff
@@ -95,16 +80,21 @@ export class ContingentLiabilityGuideStepThreeComponent implements OnInit, OnDes
...
+    }
     canDeactivate(): Promise<boolean> | boolean {
         if (this.guideService.savingStepData || !this.stepDataChanged()) {
             return true;
         }
         return this.dialog.confirm('sure');
     }
-
     stepDataChanged(): boolean {
         let dataChanged = false;
```
Comment (Komalpreet12) [inline src/app/guides/contingent-liability-guide/step-3/step-three-component.ts]: test

File: src/app/guides/long-term-debt-guide/step-2/step-two-component.ts [hunk 3]
```diff
@@ -69,6 +69,7 @@ export class LongTermDebtGuideStepTwoComponent implements OnInit, OnDestroy {
         this.stepForm.controls['longTermDebtAmount'].setValue(stepData.longTermDebtAmount);
     }
     makeFormDirtyOnError() {
+        console.log('hello');
```
[the remark this answers is not in the document]
Comment (Komalpreet12) [reply src/app/guides/long-term-debt-guide/step-2/step-two-component.ts]: i will fix this

[the remark this answers is not in the document]
Comment (Komalpreet12) [reply src/app/guides/long-term-debt-guide/step-2/step-two-component.ts]: can you please check?

File: src/app/guides/patents-and-licenses-guide/step-6/step-six-component.ts [hunk 4]
```diff
@@ -87,11 +87,19 @@ export class PatentsAndLicensesGuideStepSixComponent implements OnInit, OnDestro
...
+                if (this.patternError) {
+                    this.snackBarService.openSnackBar(this.translate.instant('VALIDATIONERROR.PATTERN'));
+                }
             } else {
                 this.snackBar.dismiss();
             }
         });
     }
+    omit_special_char(event) {
+        let k;
```
[the remark this answers is not in the document]
Comment (jorawarsingh) [reply src/app/guides/patents-and-licenses-guide/step-6/step-six-component.ts]: name is ok but as mentioned earlier looks for dups.

File: src/app/guides/staff-guide/step-2/step-two.component.ts [hunk 5]
```diff
@@ -57,7 +57,7 @@ export class StaffGuideStepTwoComponent implements OnInit, OnDestroy, StepProps
     populateFormWithStepData(stepData: any) {
         this.stepForm.controls['averageEmployees'].setValue(stepData.averageEmployees);
     }
-    omit_special_char(event) {
+    omitSpecialChar(event) {
         let k;
         k = event.charCode;  //         k = event.keyCode;  (Both can be used)
         return((k > 47 && k < 91) || (k > 96 && k < 123) || k === 8 || k === 32 || (k >= 48 && k <= 57));
```
Comment (jorawarsingh) [inline src/app/guides/staff-guide/step-2/step-two.component.ts]: dups

File: src/app/guides/todo-guides/guides-list.components.ts [hunk 6]
```diff
@@ -53,27 +53,29 @@ export class ToDoGuideListComponent implements OnInit, OnDestroy {
             if (guide.id) {
                 // this.guideService.guideList.forEach((guide: GuideDTO) => guide.id === guideId['guideId'] && (guide.completed = true));
```
Comment (jorawarsingh) [inline src/app/guides/todo-guides/guides-list.components.ts]: code formatting..

File: src/app/shared/directives/number-only.ts [hunk 7]
```diff
@@ -6,36 +6,10 @@ export class NumbersOnlyDirective {
...
-        // Ensure that it is a number and stop the keypress
-        if ((e.shiftKey || (keyCode < 48 || keyCode > 57))) {
-            e.preventDefault();
-        }
+    omitSpecialChar(event) {
+        let k;
+        k = event.charCode;  //         k = event.keyCode;  (Both can be used)
+        return((k > 44 && k < 91) || (k > 96 && k < 123) || k === 8 || k === 32 || (k >= 48 && k <= 57));
     }
 }
```
Comment (jorawarsingh) [inline src/app/shared/directives/number-only.ts]: mention char in comments above this method so another developer know which char are forbidden.
e.g
/*
*forbidden char
*$%&
*/

**Pull request merged**